Can't not run under  JDK 6

is that any jmx_exporter vesion can run under  JDK 6?
i try to run jmx_exporter in weblogic 10 with jdk6,it fail

.
```
***************************************************
*  To start WebLogic Server, use a username and   *
*  password assigned to an admin-level user.  For *
*  server administration, use the WebLogic Server *
*  console at http://hostname:port/console        *
***************************************************
starting weblogic with Java version:
java version "1.6.0_45"
Java(TM) SE Runtime Environment (build 1.6.0_45-b06)
Java HotSpot(TM) 64-Bit Server VM (build 20.45-b01, mixed mode)
Starting WLS with line:
/opt/jdk1.6.0_45/bin/java -server   -Xms8192m -Xmx8192m -XX:MaxPermSize=512m -Xmn3g  -Dweblogic.Name=AdminServer -Djava.security.policy=/u01/oracle/weblogic/wlserver_10.3/server/lib/weblogic.policy  -Dweblogic.ProductionModeEnabled=true  -Djavax.management.builder.initial=weblogic.management.jmx.mbeanserver.WLSMBeanServerBuilder -Dfile.encoding=utf-8 -javaagent:/apm/jmx_prometheus_javaagent-0.14.0.jar=8081:/apm/config.yaml  -da -Dplatform.home=/u01/oracle/weblogic/wlserver_10.3 -Dwls.home=/u01/oracle/weblogic/wlserver_10.3/server -Dweblogic.home=/u01/oracle/weblogic/wlserver_10.3/server   -Dweblogic.management.discover=true  -Dwlw.iterativeDev=false -Dwlw.testConsole=false -Dwlw.logErrorsToConsole=false -Dweblogic.ext.dirs=/u01/oracle/weblogic/patch_wls1036/profiles/default/sysext_manifest_classpath:/u01/oracle/weblogic/patch_ocp371/profiles/default/sysext_manifest_classpath  weblogic.Server
Exception in thread "main" java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
        at java.lang.reflect.Method.invoke(Method.java:597)
        at sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:323)
        at sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:338)
Caused by: java.lang.UnsupportedClassVersionError: io/prometheus/jmx/shaded/org/yaml/snakeyaml/Yaml : Unsupported major.minor version 51.0
        at java.lang.ClassLoader.defineClass1(Native Method)
        at java.lang.ClassLoader.defineClassCond(ClassLoader.java:631)
        at java.lang.ClassLoader.defineClass(ClassLoader.java:615)
        at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:141)
        at java.net.URLClassLoader.defineClass(URLClassLoader.java:283)
        at java.net.URLClassLoader.access$000(URLClassLoader.java:58)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:197)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.net.URLClassLoader.findClass(URLClassLoader.java:190)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:306)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:301)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:247)
        at io.prometheus.jmx.shaded.io.prometheus.jmx.JmxCollector.<init>(JmxCollector.java:78)
        at io.prometheus.jmx.shaded.io.prometheus.jmx.JavaAgent.premain(JavaAgent.java:29)
        ... 6 more
FATAL ERROR in native method: processing of -javaagent failed
Aborted (core dumped)
```

--- a/collector/pom.xml
+++ b/collector/pom.xml
@@ -18,7 +18,7 @@
     <dependency>
       <groupId>io.prometheus</groupId>
       <artifactId>simpleclient</artifactId>
-      <version>0.9.0</version>
+      <version>0.10.0</version>
     </dependency>
     <dependency>
       <groupId>junit</groupId>
@@ -29,7 +29,7 @@
     <dependency>
       <groupId>org.yaml</groupId>
       <artifactId>snakeyaml</artifactId>
-      <version>1.26</version>
+      <version>1.23</version>
     </dependency>
   </dependencies>
 

--- a/collector/src/main/java/io/prometheus/jmx/JmxCollector.java
+++ b/collector/src/main/java/io/prometheus/jmx/JmxCollector.java
@@ -46,7 +46,7 @@ public class JmxCollector extends Collector implements Collector.Describable {
       String help;
       boolean attrNameSnakeCase;
       boolean cache = false;
-      Type type = Type.UNTYPED;
+      Type type = Type.UNKNOWN;
       ArrayList<String> labelNames;
       ArrayList<String> labelValues;
     }
@@ -210,7 +210,12 @@ public class JmxCollector extends Collector implements Collector.Describable {
               rule.cache = (Boolean)yamlRule.get("cache");
             }
             if (yamlRule.containsKey("type")) {
-              rule.type = Type.valueOf((String)yamlRule.get("type"));
+              String t = (String)yamlRule.get("type");
+              // Gracefully handle switch to OM data model.
+              if ("UNTYPED".equals(t)) {
+                t = "UNKNOWN";
+              }
+              rule.type = Type.valueOf(t);
             }
             if (yamlRule.containsKey("help")) {
               rule.help = (String)yamlRule.get("help");

--- a/jmx_prometheus_httpserver/pom.xml
+++ b/jmx_prometheus_httpserver/pom.xml
@@ -22,7 +22,7 @@
     <dependency>
       <groupId>io.prometheus</groupId>
       <artifactId>simpleclient_httpserver</artifactId>
-      <version>0.9.0</version>
+      <version>0.10.0</version>
     </dependency>
   </dependencies>
 

--- a/jmx_prometheus_javaagent/pom.xml
+++ b/jmx_prometheus_javaagent/pom.xml
@@ -23,12 +23,12 @@
     <dependency>
       <groupId>io.prometheus</groupId>
       <artifactId>simpleclient_hotspot</artifactId>
-      <version>0.9.0</version>
+      <version>0.10.0</version>
     </dependency>
     <dependency>
       <groupId>io.prometheus</groupId>
       <artifactId>simpleclient_httpserver</artifactId>
-      <version>0.9.0</version>
+      <version>0.10.0</version>
     </dependency>
     <dependency>
       <groupId>junit</groupId>

